app/models/stolen_record.rb

```diff
@@ -277,6 +277,6 @@ def generate_alert_image
 
   # If the bike has been recovered, remove the alert_image
   def remove_outdated_alert_image
-    alert_image.remove! if date_recovered.present?
+    alert_image.remove! unless bike.reload.stolen?
```

Comment:
Why is `date_recovered` present?

Reply:
unclear. I'm looking through all the places in the code where it's set. Doesn't seem like a valid state to me.

Reply:
I agree, it doesn’t seem valid to me either. Weird that it isn’t causing other issues! Glad we’re catching it here I guess.

Reply:
There's only one, at least:
```rb
irb(main):012:0> StolenRecord.unscoped.where(current: true).where.not(date_recovered: nil).count
=> 1

```